API.md

```diff
@@ -0,0 +1,62 @@
+# SegM8 API
+
+## `class SegM8`
+
+Create an object of type `SegM8` to communicate with a chain of particular [SegM8 7-segment indicator modules](https://my.amperka.com/modules/SegM8).
+
+### `SegM8(uint8_t pinCS, uint8_t deviceCount = 1)`
+
+Constructs a new SegM8 object that uses the default hardware SPI bus.
+
+- `pinCS`: the chip select (also known as slave select) pin is used to control the shift-register latch. It is marked `CS` on the SegM8 board. Any GPIO pin of the controller may be used.
+- `deviceCount`: the number of SegM8 modules connected in a daisy-chain. If omitted, defines a single module.
+
+### `SegM8(uint8_t pinCS, uint8_t pinDI, uint8_t pinCLK, uint8_t deviceCount = 1)`
+
+Constructs a new SegM8 object that uses a software SPI bus.
+
+- `pinCS`: the chip select (also known as slave select) pin is used to control the shift-register latch. It is marked `CS` on the SegM8 module. Any GPIO pin of the controller may be used.
+- `pinDI`: the data input pin used to send data to the peripherals. It is marked `DI` on the SegM8 module. In SPI terms known as MOSI (Master Out Slave In). Any GPIO pin of the controller may be used.
+- `pinCLK`: the clock pin is used to clock pulses that synchronize data transmission generated by the master. It is marked `CLK` on the SegM8 module. Also known as SCK in SPI terms. Any GPIO pin of the controller may be used.
+- `deviceCount`: the number of SegM8 modules connected in a daisy-chain. If omitted, defines a single module.
+
+The difference between hardware and software SPI is interaction speed. Software SPI runs slower than hardware SPI but hardware SPI works only on specific board pins: `MOSI`, `MISO` (not used in SegM8) and `SCK`.
+
+### `void begin()`
+
+Initializes the given interface, prepares the board for the communication, and sets all module segments to the 'dark' state.
+Call this method before interacting with SegM8. For example, in the `setup()`.
+
+### `void startBatch(void)`
+
+The begin of output transaction. Moves current output pointer to the beginning of the daisy-chain;
+
+### `void display(int n, uint8_t width, uint8_t flags = SEGM8_ALIGN_RIGHT)`
+### `void display(unsigned int u, uint8_t width, uint8_t flags = SEGM8_ALIGN_RIGHT + SEGM8_RADIX_10)`
+### `void display(float x, uint8_t width, uint8_t precission = 1, uint8_t flags = SEGM8_ALIGN_LEFT)`
+### `void display(const char* s, uint8_t width, uint8_t flags = SEGM8_ALIGN_LEFT)`
```

Comment:
How about adding a method that also accepts a `String` type in the parameter list?

Reply:
It seemed to me that this is redundant for a number of reasons. This indicator is initially numeric, not alphanumeric. Pseudo-letters are used only to display units or special characters. There is no need to somehow handle these pseudo-strings; just set them simply as `const char*`. Is it logical?

Reply:
I thought that users use String more often: it is easier to concatenate, compare, convert to other types, etc. It might be convenient for the user to do something like: 
`display(String(some_int_value) + "-" + String(some_int_value))`.
Secondly, if the method accepts a parameter of type `String`, the user will be able to call it as `display("Some c-string")` without explicit casting. And if the method takes a parameter of type `const char *`, and the user wants to pass it a `String`, he will have to do something like `display(String_object.c_str())`.
On the other hand it really can be overkill. What do you think about this?